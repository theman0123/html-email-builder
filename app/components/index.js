export WelcomeText from './WelcomeText/WelcomeText'
export Landing from './Landing/Landing'
export Navigation from './Navigation/Navigation'