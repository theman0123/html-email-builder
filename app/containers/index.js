export HomeContainer from './Home/HomeContainer'
export LandingContainer from './Landing/LandingContainer'