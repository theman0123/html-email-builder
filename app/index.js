import React from 'react';
import { render } from 'react-dom';
import { AppContainer } from 'react-hot-loader';
import Root from './screens/Root';
import { configureStore, history } from './store/configureStore';
import './app.global.css';

const store = configureStore();

render(
  <AppContainer>
    <Root store={store} history={history} />
  </AppContainer>,
  document.getElementById('root')
);

if (module.hot) {
  module.hot.accept('./screens/Root', () => {
    const NextRoot = require('./screens/Root'); // eslint-disable-line global-require
    render(
      <AppContainer>
        <NextRoot store={store} history={history} />
      </AppContainer>,
      document.getElementById('root')
    );
  });
}
