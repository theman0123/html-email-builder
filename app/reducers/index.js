// @flow
import { combineReducers } from 'redux';
import { routerReducer as router } from 'react-router-redux';
import {reducer as burgerMenu} from 'redux-burger-menu/immutable';
import counter from './counter';
import user from './user';

const rootReducer = combineReducers({
  user,
  burgerMenu,
  counter,
  router,
});

export default rootReducer;
