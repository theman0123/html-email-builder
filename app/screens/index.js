export Home from './Home/Home';
export BuildInvoices from './BuildInvoices/BuildInvoices';