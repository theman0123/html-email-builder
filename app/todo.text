sketch the user experience...
FINISHED
0. finish redux schema--use immutable js
1. tested USER ACTIONS AND USER REDUCER; CONTACTS ACTIONS AND REDUCER;-- add flow to redux then add immutable
2. 



NEXT
--go read about 2 package.json and where to install dependencies. install dependencies 'fetch-mock' 'redux-mock-store' if needed...


a. write tests as you split off actions and reducers into their own files. -- in user



1. Start writing the tests, which can be broken down into:
-Action Tests
-Reducer Tests
-Middleware Tests
-Component Tests

2. UI for oAuth 2.0 google contacts // or error
3. UI and redux for contacts (including drag and drop)
4. UI and redux for invoice editor
5. connect redux to python3
6. send email with python3
7. show confirmation of email sent //or error 