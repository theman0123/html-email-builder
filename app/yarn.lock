# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


amdefine@>=0.0.4:
  version "1.0.1"
  resolved "https://registry.yarnpkg.com/amdefine/-/amdefine-1.0.1.tgz#4a5282ac164729e93619bcfd3ad151f817ce91f5"

asap@~2.0.3:
  version "2.0.6"
  resolved "https://registry.yarnpkg.com/asap/-/asap-2.0.6.tgz#e50347611d7e690943208bbdafebcbc2fb866d46"

ast-transform@0.0.0:
  version "0.0.0"
  resolved "https://registry.yarnpkg.com/ast-transform/-/ast-transform-0.0.0.tgz#74944058887d8283e189d954600947bc98fe0062"
  dependencies:
    escodegen "~1.2.0"
    esprima "~1.0.4"
    through "~2.3.4"

ast-types@^0.7.0:
  version "0.7.8"
  resolved "https://registry.yarnpkg.com/ast-types/-/ast-types-0.7.8.tgz#902d2e0d60d071bdcd46dc115e1809ed11c138a9"

browser-resolve@^1.8.1:
  version "1.11.2"
  resolved "https://registry.yarnpkg.com/browser-resolve/-/browser-resolve-1.11.2.tgz#8ff09b0a2c421718a1051c260b32e48f442938ce"
  dependencies:
    resolve "1.1.7"

browserify-optional@^1.0.0:
  version "1.0.1"
  resolved "https://registry.yarnpkg.com/browserify-optional/-/browserify-optional-1.0.1.tgz#1e13722cfde0d85f121676c2a72ced533a018869"
  dependencies:
    ast-transform "0.0.0"
    ast-types "^0.7.0"
    browser-resolve "^1.8.1"

classnames@^2.1.1:
  version "2.2.5"
  resolved "https://registry.yarnpkg.com/classnames/-/classnames-2.2.5.tgz#fb3801d453467649ef3603c7d61a02bd129bde6d"

core-js@^1.0.0:
  version "1.2.7"
  resolved "https://registry.yarnpkg.com/core-js/-/core-js-1.2.7.tgz#652294c14651db28fa93bd2d5ff2983a4f08c636"

encoding@^0.1.11:
  version "0.1.12"
  resolved "https://registry.yarnpkg.com/encoding/-/encoding-0.1.12.tgz#538b66f3ee62cd1ab51ec323829d1f9480c74beb"
  dependencies:
    iconv-lite "~0.4.13"

escodegen@~1.2.0:
  version "1.2.0"
  resolved "https://registry.yarnpkg.com/escodegen/-/escodegen-1.2.0.tgz#09de7967791cc958b7f89a2ddb6d23451af327e1"
  dependencies:
    esprima "~1.0.4"
    estraverse "~1.5.0"
    esutils "~1.0.0"
  optionalDependencies:
    source-map "~0.1.30"

esprima@~1.0.4:
  version "1.0.4"
  resolved "https://registry.yarnpkg.com/esprima/-/esprima-1.0.4.tgz#9f557e08fc3b4d26ece9dd34f8fbf476b62585ad"

estraverse@~1.5.0:
  version "1.5.1"
  resolved "https://registry.yarnpkg.com/estraverse/-/estraverse-1.5.1.tgz#867a3e8e58a9f84618afb6c2ddbcd916b7cbaf71"

esutils@~1.0.0:
  version "1.0.0"
  resolved "https://registry.yarnpkg.com/esutils/-/esutils-1.0.0.tgz#8151d358e20c8acc7fb745e7472c0025fe496570"

eve@~0.5.1:
  version "0.5.4"
  resolved "https://registry.yarnpkg.com/eve/-/eve-0.5.4.tgz#67d080b9725291d7e389e34c26860dd97f1debaa"

fbjs@^0.8.16:
  version "0.8.16"
  resolved "https://registry.yarnpkg.com/fbjs/-/fbjs-0.8.16.tgz#5e67432f550dc41b572bf55847b8aca64e5337db"
  dependencies:
    core-js "^1.0.0"
    isomorphic-fetch "^2.1.1"
    loose-envify "^1.0.0"
    object-assign "^4.1.0"
    promise "^7.1.1"
    setimmediate "^1.0.5"
    ua-parser-js "^0.7.9"

iconv-lite@~0.4.13:
  version "0.4.19"
  resolved "https://registry.yarnpkg.com/iconv-lite/-/iconv-lite-0.4.19.tgz#f7468f60135f5e5dad3399c0a81be9a1603a082b"

immutable@^3.8.2:
  version "3.8.2"
  resolved "https://registry.yarnpkg.com/immutable/-/immutable-3.8.2.tgz#c2439951455bb39913daf281376f1530e104adf3"

is-stream@^1.0.1:
  version "1.1.0"
  resolved "https://registry.yarnpkg.com/is-stream/-/is-stream-1.1.0.tgz#12d4a3dd4e68e0b79ceb8dbc84173ae80d91ca44"

isomorphic-fetch@^2.1.1:
  version "2.2.1"
  resolved "https://registry.yarnpkg.com/isomorphic-fetch/-/isomorphic-fetch-2.2.1.tgz#611ae1acf14f5e81f729507472819fe9733558a9"
  dependencies:
    node-fetch "^1.0.1"
    whatwg-fetch ">=0.10.0"

js-tokens@^3.0.0:
  version "3.0.2"
  resolved "https://registry.yarnpkg.com/js-tokens/-/js-tokens-3.0.2.tgz#9866df395102130e38f7f996bceb65443209c25b"

loose-envify@^1.0.0, loose-envify@^1.3.1:
  version "1.3.1"
  resolved "https://registry.yarnpkg.com/loose-envify/-/loose-envify-1.3.1.tgz#d1a8ad33fa9ce0e713d65fdd0ac8b748d478c848"
  dependencies:
    js-tokens "^3.0.0"

node-fetch@^1.0.1:
  version "1.7.3"
  resolved "https://registry.yarnpkg.com/node-fetch/-/node-fetch-1.7.3.tgz#980f6f72d85211a5347c6b2bc18c5b84c3eb47ef"
  dependencies:
    encoding "^0.1.11"
    is-stream "^1.0.1"

object-assign@^4.1.0, object-assign@^4.1.1:
  version "4.1.1"
  resolved "https://registry.yarnpkg.com/object-assign/-/object-assign-4.1.1.tgz#2109adc7965887cfc05cbbd442cac8bfbb360863"

promise@^7.1.1:
  version "7.3.1"
  resolved "https://registry.yarnpkg.com/promise/-/promise-7.3.1.tgz#064b72602b18f90f29192b8b1bc418ffd1ebd3bf"
  dependencies:
    asap "~2.0.3"

prop-types@^15.5.8:
  version "15.6.1"
  resolved "https://registry.yarnpkg.com/prop-types/-/prop-types-15.6.1.tgz#36644453564255ddda391191fb3a125cbdf654ca"
  dependencies:
    fbjs "^0.8.16"
    loose-envify "^1.3.1"
    object-assign "^4.1.1"

react-burger-menu@^2.3.0:
  version "2.3.0"
  resolved "https://registry.yarnpkg.com/react-burger-menu/-/react-burger-menu-2.3.0.tgz#a5849d085115457e8b46516bb4dd9432e09fc805"
  dependencies:
    browserify-optional "^1.0.0"
    classnames "^2.1.1"
    eve "~0.5.1"
    prop-types "^15.5.8"
    snapsvg-cjs "0.0.6"

resolve@1.1.7:
  version "1.1.7"
  resolved "https://registry.yarnpkg.com/resolve/-/resolve-1.1.7.tgz#203114d82ad2c5ed9e8e0411b3932875e889e97b"

setimmediate@^1.0.5:
  version "1.0.5"
  resolved "https://registry.yarnpkg.com/setimmediate/-/setimmediate-1.0.5.tgz#290cbb232e306942d7d7ea9b83732ab7856f8285"

snapsvg-cjs@0.0.6:
  version "0.0.6"
  resolved "https://registry.yarnpkg.com/snapsvg-cjs/-/snapsvg-cjs-0.0.6.tgz#3b2f56af2573d3d364c3ed5bf8885745f4d2dde1"
  dependencies:
    snapsvg "0.5.1"

snapsvg@0.5.1:
  version "0.5.1"
  resolved "https://registry.yarnpkg.com/snapsvg/-/snapsvg-0.5.1.tgz#0caf52c79189a290746fc446cc5e863f6bdddfe3"
  dependencies:
    eve "~0.5.1"

source-map@~0.1.30:
  version "0.1.43"
  resolved "https://registry.yarnpkg.com/source-map/-/source-map-0.1.43.tgz#c24bc146ca517c1471f5dacbe2571b2b7f9e3346"
  dependencies:
    amdefine ">=0.0.4"

through@~2.3.4:
  version "2.3.8"
  resolved "https://registry.yarnpkg.com/through/-/through-2.3.8.tgz#0dd4c9ffaabc357960b1b724115d7e0e86a2e1f5"

ua-parser-js@^0.7.9:
  version "0.7.17"
  resolved "https://registry.yarnpkg.com/ua-parser-js/-/ua-parser-js-0.7.17.tgz#e9ec5f9498b9ec910e7ae3ac626a805c4d09ecac"

whatwg-fetch@>=0.10.0:
  version "2.0.3"
  resolved "https://registry.yarnpkg.com/whatwg-fetch/-/whatwg-fetch-2.0.3.tgz#9c84ec2dcf68187ff00bc64e1274b442176e1c84"
