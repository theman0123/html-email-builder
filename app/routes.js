/* eslint flowtype-errors/show-errors: 0 */
import React from 'react';
import { Switch, Route } from 'react-router';
import App from './screens/App';
import { Home, BuildInvoices } from 'screens';

export default () => (
  <App>
    <Switch>
      <Route exact path="/" component={Home} />
      <Route path="/build-invoices" component={BuildInvoices} />
    </Switch>
  </App>
);
