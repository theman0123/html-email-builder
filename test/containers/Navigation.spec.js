import React from 'react';
import renderer from 'react-test-renderer'
import { Navigation } from '../../app/components';
import { action as toggleMenu } from 'redux-burger-menu/immutable';
import { slide as Menu } from 'react-burger-menu';
import { decorator as reduxBurgerMenu } from 'redux-burger-menu/immutable';
import { configureStore, history } from '../../app/store/configureStore';


describe('NAVIGATION COMPONENT', () => {
  describe('Navigation icon should be visible', () => {
//    const store = configureStore();
    
//    it('Match previous visible styling', () => {
//      const wrapper = shallow(<Navigation />);
//     figure out a way to test if component is visible -> look at testing css console.log(wrapper.find('.burgerButtonMenu'), styles.burgerButtonClassName)
//      expect(wrapper.props()).toMatchSnapshot();
//    });
  });
  describe(':::store.burgerMenu represents bugerMenu component:::', () => {
    const store = configureStore();
    const wrapper = shallow(<Navigation store={store}/>);
    const initialBM = store.getState().burgerMenu.get('isOpen');
//    const initialLocation = store.getState().router.location;
    
    it('should be connected to store', () => {
      expect(initialBM).toMatchSnapshot();
    });
    it('should display current route from props.selected', () => {
      expect(wrapper.prop('selected')).toEqual('Home');
    });
    it('should have handleClick', () => {
      const handleClick = wrapper.dive().instance().handleClick;
      const handleExists = handleClick !== "undefined";
      expect(handleExists).toBe(true);
    });
    it('should close nav menu with onClick/handleClick', () => {
      const closed = initialBM === false;
      store.dispatch(toggleMenu(true));
      
      const open = store.getState().burgerMenu.get('isOpen');
      wrapper.dive().find('Link').first().simulate('click');
      const afterSelect = 'nothing';
      console.log(closed, open, afterSelect, wrapper.dive().find('Link').first().simulate('click'));
//      const openSelectLinkClose =  closed && open && afterSelect; 
      //check store.menuOpen should be true
      //click
      //check store.menuOpen -> should be false
      expect(wrapper.dive().find('Link').first().simulate('click'));
    });
  });
});
