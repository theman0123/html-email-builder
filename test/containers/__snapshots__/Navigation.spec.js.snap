// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NAVIGATION COMPONENT :::store.burgerMenu represents bugerMenu component::: should be connected to store 1`] = `false`;
